src/agent/config.ts

```diff
@@ -119,54 +119,86 @@ export interface DebugAgentConfig extends common.AuthenticationConfig {
      * data and subsequently the network traffic. A value of 0 disables the
      * limit.
      */
-    maxDataSize: number;
+    maxDataSize?: number;
 
     /**
      * To limit the size of the buffer, we truncate long strings. A value of 0
      * disables truncation.
      */
-    maxStringLength: number;
+    maxStringLength?: number;
   };
 
   /**
    * options affecting log points.
    */
-  log: {
+  log?: {
     /**
      * The maximum number of logs to record per second per logpoint.
      */
-    maxLogsPerSecond: number;
+    maxLogsPerSecond?: number;
 
     /**
      * Number of seconds to wait after the `maxLogsPerSecond` rate is hit before
      * logging resumes per logpoint.
      */
-    logDelaySeconds: number;
+    logDelaySeconds?: number;
   };
 
   /**
    * These configuration options are for internal  experimentation only.
    */
-  internal: {
-    registerDelayOnFetcherErrorSec: number; maxRegistrationRetryDelay: number;
+  internal?: {
+    registerDelayOnFetcherErrorSec?: number;
+    maxRegistrationRetryDelay?: number;
   };
 
   /**
    * Used by tests to force loading of a new agent if one exists already
    */
-  forceNewAgent_: boolean;
+  forceNewAgent_?: boolean;
 
   /**
    * Uses by tests to cause the start() function to return the debuglet.
    */
+  testMode_?: boolean;
+}
+
+export interface ResolvedDebugAgentConfig extends DebugAgentConfig {
```

Comment:
Do [`Partial<T>`](https://www.typescriptlang.org/docs/handbook/advanced-types.html) types work here?

Reply:
Nice.  I wasn't aware of `Partial<T>` types.  I've updated the code to use them.